flag.go

```diff
@@ -825,16 +834,20 @@ func (f *FlagSet) parseSingleShortArg(shorthands string, args []string) (outShor
 		err = f.failf("flag needs an argument: %q in -%s", c, shorthands)
 		return
 	}
-	err = f.setFlag(flag, value, shorthands)
-	return
+
+	origVal := value
+	if defVal {
+		origVal = ""
+	}
+	return outShorts, outArgs, fn(flag, value, shorthands, "-"+shorthands[:1], origVal)
```

Comment:
Wow, I've been letting this sit around forever. Thinking about it I think I've kinda changed my mind. I think I only want to pass `*flag.Flag` and `value` to the function. This would create a very simple reasonable function for people to use. It would, however, mean that you cannot use this method to exactly recreate the original command line. 'Recreating' the command line would ALWAYS return the long name and would ALWAYS include the value. So something like
`-v 10` would get 'recreated' and `--verbose=10`.  Or `-h` would get turned into `--help=true`.  I personally think that is reasonable, simple, and gives a very good API.

What do you think @xilabao ?